src/deploy/ecr.py

```diff
@@ -8,50 +8,25 @@
 from .commands import cmd
 
 
-class EcrImage:
-    """
-    Convenience wrapper around a response from the ECR DescribeImages API.
-    """
-    def __init__(self, ecr_base_uri, repository_name, tag, describe_images_resp):
-        if not describe_images_resp["imageDetails"]:
-            raise EcrError(f"No matching images found for {repository_name}:{tag}!")
-
-        if len(describe_images_resp["imageDetails"]) > 1:
-            raise EcrError(f"Multiple matching images found for {repository_name}:{tag}!")
-
-        self._image_details = describe_images_resp["imageDetails"][0]
-        self.ecr_base_uri = ecr_base_uri
-        self.repository_name = repository_name
-        self.tag = tag
-
-    @property
-    def tags(self):
-        return set(self._image_details["imageTags"])
-
-    def ref_uri(self):
-        ref_tags = {t for t in self.tags if t.startswith("ref.")}
-
-        if not ref_tags:
-            raise EcrError(f"No matching ref tags found for {self.repository_name}:{self.tag}!")
-
-        # It's possible to get multiple ref tags if the same image is published
-        # at different Git commits, but there are no code changes for this image
-        # between the two commits.  If so, choose one arbitrarily.
-        ref = ref_tags.pop()
+def create_client(*, account_id, region_name, role_arn):
```

Comment:
I notice a pattern of putting some function in the top level module space and  some things in a class. I find it a bit weird because i'm used to things more in OOPland. What's the rationale your using to determine where things should remain in class and where they should be a "module function" (is that the right terminology).

Reply:
I tend to go for module functions if they're standalone pieces that wouldn't be used with anything else. Part of this is reducing the amount of shared state that's required; if all you need to pass around is the client, I think it's fine to do that as a function (and makes testing slightly simpler).

e.g. the Ecs and Ecr classes contain bits of cached state that are only relevant to one function; splitting off functions that don't need them makes it clearer where the boundaries are.

But I'm not sure there are hard and fast rules.